notebook/services/kernels/kernelmanager.py

```diff
@@ -310,37 +310,62 @@ async def restart_kernel(self, kernel_id, now=False):
         await maybe_future(self.pinned_superclass.restart_kernel(self, kernel_id, now=now))
         kernel = self.get_kernel(kernel_id)
         # return a Future that will resolve when the kernel has successfully restarted
-        channel = kernel.connect_shell()
+        shell_channel = kernel.connect_shell()
+        iopub_channel = kernel.connect_iopub()
```

Comment:
Really, we should use the iopub channel from `zmqhandler.channels`, which we don't have access to here.